Userland/Libraries/LibSoftGPU/Device.cpp

```diff
@@ -991,6 +991,9 @@ ALWAYS_INLINE void Device::shade_fragments(PixelQuad& quad)
             quad.out_color.set_z(mix(quad.out_color.z(), texel.z(), dst_alpha));
             break;
         }
+        case TextureEnvMode::Add:
+            quad.out_color = quad.out_color + texel;
```

Comment:
I think we should leave alpha alone in most of the cases for `GL_ADD`?

Reply:
Actually it's more complicated and depends on the texture format 🙃 See the wide table here: https://www.khronos.org/registry/OpenGL-Refpages/gl2.1/xhtml/glTexEnv.xml

Reply:
> I think we should leave alpha alone in most of the cases for `GL_ADD`?

You're right, I misread this table completely.... It would help if it rendered better haha.

@sunverwerth Shall I just add the `rgb` components for now?

Reply:
@Quaker762 I think multiplyig the alpha components and adding a FIXME for GL_INTENSITY and GL_ALPHA texture format would be good for now.